Types.h

```diff
@@ -5,6 +5,35 @@
 
 #include <MY17_Can_Library.h>
 
+// All names in the enum are taken from the NDrive manual
+typedef enum {
```

Comment:
I don't see why this enum is necessary. The current approach opens you up to bugs (what happens if you accidentally pass in a I_CMD to the struct instead of a CAN_MC_REG_CURRENT_CMD) and requires you to write that whole ugly switch statement in Mc_Request.cpp that solely functions to convert between these two.

If the thing you're concerned about is the length of the names in the existing REGID enum from the CAN library, you can solve this with creating a #define for each of them rather than having to have a new enum. 

The only benefit of the current approach that I can see is that you can "move" to the next transmit type in the current state by just adding one to the last setting, which is actually pretty dank. However you sadly don't take advantage of this :(